polaris-react/src/components/Badge/Badge.tsx

```diff
@@ -52,6 +52,7 @@ export function Badge({
     styles.Badge,
     status && styles[variationName('status', status)],
     icon && styles.icon,
+    // TODO: remove support for the size prop in the next major release
```

Comment:
This is a bit weird because we're removing the styling for the small prop. Can we deprecate the size prop in this PR? 

Something like:

```tsx
  /** @deprecated Use the default size. */
  size?: Size;